Pairing api

rainrivas opened the pull request:
Laying the foundation for #31 ; I don't think 31 is done yet, but it's a start.

rainrivas committed:
Setting up a very basic set of api routes

rainrivas committed:
Adding basic logic for /pairing route. Modifying Donor and Family schema
to make references to each otehr for matching and adding default values
to it. /pairing should find all donors and filter them based on budget then sort budgets in descending order.

rainrivas committed:
Adding organizer to the donor and family models

rainrivas committed:
Adding pseudo logic for /paring/balance route

rainrivas committed:
Adding notes to /pairing/balance to remind us to use the readWishlist() method

psyduc commented on server/routes/api.js:
```diff
@@ -1,9 +1,103 @@
+  let minBudget = req.params.budget*0.9;
+
+  Donor
+    .where('budget').gte(maxBudget).lte(minBudget)
```
budget should be lte maxBudget and gte minBudget

psyduc commented on server/routes/api.js:
```diff
@@ -1,9 +1,103 @@
+    res.status(500).json({ message: err });
+  }
+  else {
+   pres.status(200).json({ money });
```
pres?

psyduc reviewed (changes requested):
switch the min and max budget variables and fix the typo

jwu910 commented on server/routes/api.js:
```diff
@@ -1,9 +1,103 @@
+const Donor = require('../models/donor');
+const Family = require('../models/family');
+// wishlist = Family.readWishlist();
+mongoose.connect('mongodb://localhost/aaf_local');
```
When we move to a production environment. Will we still be using mongoose? Also a new DB instance will be created per AAF instance, right?

Regardless, this will change to some relative path instead of localhost right?

phuchle replied:
Yes, mongoose will carry over into production.  The URL we are using for development right now will change.  Later on, either I or Efrain can set the mongoDB URL to point to a `process.env` variable for production or a local URL for development.

jwu910 replied:
Ah, right on. So Mongoose will be the one creating each new schema(?) or data ojbect when a new family is added or a new donor registers, etc?

phuchle replied:
Not sure what your question is getting at.  Mongoose is a tool to model the data that will be inserted into MongoDB and provides an easy API to interact with MongoDB.

jwu910 replied:
I think i am mixing up my vocabulary. I'll just keep reading. =] Its coming together though. 👍

jwu910 reviewed (changes requested):
Just same as duc's changes. Cool stuff.

rainrivas committed:
Correcting issues with api.js

Switching the min/max budgets in the LTE/GTE query and removing the p from "res"
